### DISCUSSION TITLE
Expose the setup-frontend or a status / override frontend over https

### DISCUSSION THREAD

**Discussion by frodeheg** [frodeheg/no.sparegris#36] [Polls]
In Issue #30 an interesting proposal popped up, making the app act as a https server and then expose the setup frontend or a status / override page or any other page for that matter without the limitations of the Homey UI.

What do you think?

**Comment by dooniem** [comment]
Could we list advantages with having a https server?
I voted yes, but I'm not sure if I'd use it / understand all advantages this would bring to the table.
It can't be just easier maintenance of ACs?

Here are things I'm curious if will become true or not. Hopefully other people will list what they see / hope to get as an advantage

- Quicker access to specific settings, I guess this is a given
- If PB in the future logs and timestamp what it's doing to your home, this log could be read and handled through other apps/HTTP GET flows and hence get quicker access through shortcuts or publish them to an iPad or something crazy?
- Make flows to enable and disable PBs control over a specific device or change a setting? But I guess this is why the app is still under development, making http send unnecessary?

**Reply by frodeheg** [reply]
At first, most of the work would be implementing this in a secure way, and probably just exposing the settings page as this in fact is just a web-page already.

Then, later I would probably let you customize what would be shown. This would be very nice to just put the web-page into a widget like https://play.google.com/store/apps/details?id=com.urysoft.widgery&hl=en&gl=US to have it on your android home screen.

**Comment by frodeheg** [comment]
Well, have you seen, app widgets has already been available for 2 years: https://support.homey.app/hc/en-us/articles/360012168019-Starting-a-Flow-from-an-iOS-Android-widget That makes it a bit more interesting with overrides as flows as you could have the override already in your home screen... And a little less interesting with control over https?
